File: Initial_Python_Version_Backend/library/text_to_voice.py
from library import concurrent_functions_organizer
import time

class VoiceProducer:
    def __init__(self, runner_reference):
        self.runner = runner_reference
        self.c_voice = concurrent_functions_organizer.ConcurrentOrgnizer()
    
    def update(self):
        return self.try_get_voice_response()

    # request_content is required to be in format {"content":"xyz", ....}
    def send_text_to_voice_request(self, request_content, start_time = 0):
        if start_time == 0:
            start_time = time.time()
        return self.c_voice.start_request_response_customized_start_time(request_content, self.runner.voice_ai.request_voice, start_time, sole_exist=False) # change the request func
    
    def try_get_voice_response(self):
        return self.c_voice.get_response(False)

File: React_Frontend/src/system_runner/text_to_voice.js
import { ConcurrentOrgnizer } from './concurrent_runner.js';

export class VoiceProducer {
  constructor(runnerReference) {
    this.runner = runnerReference;
    this.cVoice = new ConcurrentOrgnizer();
  }

  update() {
    return this.tryGetVoiceResponse();
  }

  // requestContent is required to be in format {"content":"xyz", ....}
  sendTextToVoiceRequest(requestContent, startTime = 0) {
    if (startTime === 0) {
      startTime = Date.now() / 1000;
    }
    return this.cVoice.startRequestResponseCustomizedStartTime(requestContent, this.runner.voiceAi.requestVoice.bind(this.runner.voiceAi.requestVoice), startTime, false); // change the request func
  }

  tryGetVoiceResponse() {
    return this.cVoice.getResponse(false);
  }
}